File: network/src/main/java/com/app/network/interceptors/ConnecitivityInterceptor.java
package com.app.network.interceptors;


import com.app.jenkins_common.utils.NetworkUtils;

import java.io.IOException;

import javax.inject.Inject;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public final class ConnecitivityInterceptor implements Interceptor {
  private final NetworkUtils connectivityManager;

  @Inject
  public ConnecitivityInterceptor(final NetworkUtils networkUtils) {
    this.connectivityManager = networkUtils;
  }


  @Override
  public Response intercept(final Chain chain) throws IOException {
    if (connectivityManager.isOnline()) {
      return null;
    }
    final Request.Builder builder = chain.request().newBuilder();
    return chain.proceed(builder.build());
  }
}

File: network/src/main/java/com/app/network/interceptors/ConnecitivityInterceptor.kt
package com.app.network.interceptors


import com.app.jenkins_common.utils.NetworkUtils
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import javax.inject.Inject

class ConnecitivityInterceptor @Inject
constructor(private val connectivityManager: NetworkUtils) : Interceptor {


    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response? {
        if (connectivityManager.isOnline) {
            return null
        }
        val builder = chain.request().newBuilder()
        return chain.proceed(builder.build())
    }
}
